src/topology/fanout.rs

```diff
@@ -129,7 +129,30 @@ impl Sink for Fanout {
                 Ok(Async::NotReady) => {
                     all_complete = false;
                 }
-                Err(()) => self.handle_sink_error()?,
+                Err(()) => self.handle_sink_error(i)?,
+            }
+        }
+
+        if all_complete {
+            Ok(Async::Ready(()))
+        } else {
+            Ok(Async::NotReady)
+        }
+    }
+
+    fn close(&mut self) -> Poll<(), Self::SinkError> {
+        self.process_control_messages();
+
+        let mut all_complete = true;
+
+        for i in 0..self.sinks.len() {
+            let (_name, sink) = &mut self.sinks[i];
+            match sink.close() {
```

Comment:
Should we remove sinks that have returned `Ready` from `close` to avoid calling it after they're already closed? It seems like that could potentially cause issues (futures don't like being polled after they're complete), but I don't see anything in the docs saying not to do it. So maybe not a big deal?

We could also reset and reuse `self.i`, since it is not allowed to call `start_send` or `poll_complete` after calling `close`.

Reply:
I've not seen anything in the docs to indicate this as a limitation so am inclined to avoid the added complication. In most cases it should behave the same as `poll_complete` where it just returns Ready again